src/ssl.c

```diff
@@ -12005,6 +12005,166 @@ static int wolfSSL_remove_ciphers(char* list, int sz, const char* toRemove)
+/* @param list   suites list that user wants to update                      */
+/* @return suites list on successs, otherwise NULL                          */
+static char* buildEnabledCipherList(WOLFSSL_CTX* ctx, Suites* suites,
+           int onlytlsv13suites, const char* list)
+{
+    word32 idx = 0;
+    word32 listsz = 0;
+    word32 len = 0;
+    word32 ianasz = 0;
+    const char* enabledcs = NULL;
+    char* locallist = NULL;
+    char* head = NULL;
+    byte cipherSuite0;
+    byte cipherSuite;
+    
+    /* sanity check */
+    if (ctx == NULL || suites == NULL || list == NULL)
+        return NULL;
+        
+    if (!suites->setSuites)
+        return NULL;
+    
+    listsz = (word32)XSTRLEN(list);
+    
+    /* calculate necessary buffer length */
+    for(idx = 0; idx < suites->suiteSz; idx++) {
+        
+        cipherSuite0 = suites->suites[idx];
+        cipherSuite  = suites->suites[++idx];
+        
+        if (onlytlsv13suites && cipherSuite0 == TLS13_BYTE) {
+            enabledcs = GetCipherNameInternal(cipherSuite0, cipherSuite);
+        } else if (!onlytlsv13suites && cipherSuite0 != TLS13_BYTE) {
+            enabledcs = GetCipherNameInternal(cipherSuite0, cipherSuite);
+        } else
+            continue;
+        
+        if (XSTRNCMP(enabledcs, "None", XSTRLEN(enabledcs)) != 0) {
+            len += (word32)XSTRLEN(enabledcs) + 2;
+        }
+    }
+    
+    len += listsz + 2;
+    
+    /* build string */
+    if (len > 0) {
+        locallist = (char*)XMALLOC(len, ctx->heap,
+                                           DYNAMIC_TYPE_TMP_BUFFER);
+        /* sanity check */
+        if (!locallist)
+            return NULL;
+        
+        XMEMSET(locallist, 0, len);
+        
+        head = locallist;
+        
+        if (!onlytlsv13suites)
+        {
+            /* always tls13 suites in the head position */
+            XSTRNCPY(locallist, list, len);
+            locallist += listsz;
+            *locallist++ = ':';
+            *locallist = 0;
+            len -= listsz + 1;
+        }
+        
+        for(idx = 0; idx < suites->suiteSz; idx++) {
+            cipherSuite0 = suites->suites[idx];
+            cipherSuite  = suites->suites[++idx];
+            
+            if (onlytlsv13suites && cipherSuite0 == TLS13_BYTE) {
+                enabledcs = GetCipherNameInternal(cipherSuite0, cipherSuite);
+            } else if (!onlytlsv13suites && cipherSuite0 != TLS13_BYTE) {
+                enabledcs = GetCipherNameInternal(cipherSuite0, cipherSuite);
+            } else
+                continue;
+            
+            ianasz = (int)XSTRLEN(enabledcs);
+            if (ianasz + 1 < len) {
+                XSTRNCPY(locallist, enabledcs, len);
+                locallist += ianasz;
+                
+                *locallist++ = ':';
+                *locallist = 0;
+                len -= ianasz + 1;
+            }
+            else{
```

Comment:
Cipher suite should have been validated above when calculating length.

Reply:
Hello @SparkiDev 
This means that cipher suites in the "list" that user sets should be validated? I think that it is done in SetCipherList(). Therefore, it just passes the list as is. I should miss something. Please let me know more detail.